src/Service/ThemeResolver.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace SilverStripe\Subsites\Service;
+
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\Subsites\Model\Subsite;
+use SilverStripe\View\SSViewer;
+
+class ThemeResolver
+{
+    use Injectable;
+    use Configurable;
+
+    /**
+     * Cascading definitions for themes, keyed by the name they should appear under in the CMS. For example:
+     *
+     * [
+     *   'theme-1' => [
+     *     '$public',
+     *     'starter',
+     *     '$default',
+     *   ],
+     *   'theme-2' => [
+     *     'custom',
+     *     'watea',
+     *     'starter',
+     *     '$public',
```

Comment:
The public entry would always be at the start

Reply:
Eh? You can choose.